### PR TITLE
fix: citgm

### PR DESCRIPTION
**Pull request by simone-sanfratello**
Following https://github.com/nodejs/citgm/pull/973

Tested locally on node 21 nightly

test files:
- `custom-parser.1.test.js`
- `listen.test.js`
- `plugin.test.js`
- `route.test.js`
- `stream.test.js`

Have been splitted because they contained too many tests, that hit timeout on slow citgm machines

Note before merge:

- remove debugging `console.log`s
- restore `npm test` as it was
- make sure `citgm` script is been used on citgm, must be in `lookup.json` file or citgm settings

Note to future me: run it as `citgm git+https://github.com/simone-sanfratello/fastify#fix/citgm -l lookup.json`

lookup.json
```json
{
  "useGitClone": true,
  "scripts": ["citgm"]
}
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -109,7 +109,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [14, 16, 18, 20]
+        node-version: [14, 16, 18, 20, 21]
```
Comment (simone-sanfratello) [inline .github/workflows/ci.yml:112]: tmp

**Comment by simone-sanfratello** [comment]
Note before merge:
- remove debugging console.log s
- restore npm test
- make sure citgm script is been used on citgm

File: package.json [hunk 2]
```diff
@@ -26,10 +26,10 @@
     "test:validator:integrity": "npm run build:validation && git diff --quiet --ignore-all-space --ignore-blank-lines --ignore-cr-at-eol lib/error-serializer.js && git diff --quiet --ignore-all-space --ignore-blank-lines --ignore-cr-at-eol lib/configValidator.js",
     "test:typescript": "tsc test/types/import.ts && tsd",
     "test:watch": "npm run unit -- -w --no-coverage-report -R terse",
-    "unit": "c8 tap",
+    "unit": "c8 tap --jobs=1",
```
Comment (simone-sanfratello) [inline package.json:29]: to be reset

Comment (mcollina) [reply package.json:29]:
```suggestion
    "unit": "c8 tap",
```

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
We have citgm passing now with this PR.

File: test/build-certificate.js [hunk 3]
```diff
@@ -1,6 +1,95 @@
 'use strict'
 
-const selfCert = require('self-cert')
+const os = require('os')
+const forge = require('node-forge')
```
Comment (jsumners) [inline test/build-certificate.js:4]: I missed this and am now seeing it in #5757. Why did we embed this code? @simone-sanfratello @mcollina

**Pull request merged**